Remotion/Data/DomainObjects.Security.UnitTests/SecurityClientTransactionExtensionTests/TestHelper.cs

```diff
@@ -144,85 +130,161 @@ public void AddExtension (IClientTransactionExtension extension)
     public void ExpectObjectSecurityStrategyHasAccess (SecurableObject securableObject, Enum accessTypeEnum, HasAccessDelegate doDelegate)
     {
       IObjectSecurityStrategy objectSecurityStrategy = securableObject.GetSecurityStrategy();
-      Expect.Call(
-          objectSecurityStrategy.HasAccess(
-              Arg.Is(_mockSecurityProvider),
-              Arg.Is(_stubUser),
-              Arg<IReadOnlyList<AccessType>>.List.Equal(new[] { AccessType.Get(accessTypeEnum) })))
-          .WhenCalled(mi => CheckTransaction())
-          .Do(doDelegate).Repeat.Once();
+      Mock.Get(objectSecurityStrategy)
+          .Setup(_ => _.HasAccess(
+              _mockSecurityProvider.Object,
+              _stubUser.Object,
+              new[] { AccessType.Get(accessTypeEnum) }))
+          .Callback((ISecurityProvider securityProvider, ISecurityPrincipal principal, IReadOnlyList<AccessType> requiredAccessTypes) => CheckTransaction())
+          .Returns(doDelegate)
+          .Verifiable();
     }
 
     public void ExpectObjectSecurityStrategyHasAccess (SecurableObject securableObject, Enum accessTypeEnum, bool returnValue)
     {
       IObjectSecurityStrategy objectSecurityStrategy = securableObject.GetSecurityStrategy();
-      Expect.Call(
-          objectSecurityStrategy.HasAccess(
-              Arg.Is(_mockSecurityProvider),
-              Arg.Is(_stubUser),
-              Arg<IReadOnlyList<AccessType>>.List.Equal(new[] { AccessType.Get(accessTypeEnum) })))
-          .WhenCalled(mi => CheckTransaction())
-          .WhenCalled(mi => CheckTransaction())
-          .Return(returnValue).Repeat.Once();
+      Mock.Get(objectSecurityStrategy)
+          .Setup(_ => _.HasAccess(
+              _mockSecurityProvider.Object,
+              _stubUser.Object,
+              new[] { AccessType.Get(accessTypeEnum) }))
+          .Callback(
+              (ISecurityProvider securityProvider, ISecurityPrincipal principal, IReadOnlyList<AccessType> requiredAccessTypes) =>
+              {
+                CheckTransaction();
+                CheckTransaction();
+              })
+          .Returns(returnValue)
+          .Verifiable();
     }
 
-    public void ExpectFunctionalSecurityStrategyHasAccess (Type securableObjectType, Enum accessTypeEnum, HasStatelessAccessDelegate doDelegate)
+    public void ExpectObjectSecurityStrategyHasAccess (MockSequence sequence, SecurableObject securableObject, Enum accessTypeEnum, bool returnValue)
```

Comment:
- [ ] Rework to pass Mock<SecurableObject> instead of SecurableObject
- [ ] Check if we can use a single method for both Sequence and no-Sequence.